controllers/machine_controller_phases.go

```diff
@@ -172,6 +173,10 @@ func (r *MachineReconciler) reconcileBootstrap(ctx context.Context, cluster *clu
 	}
 	bootstrapConfig := externalResult.Result
 
+	// Report a summary of current status of the bootstrap object defined for this machine.
+	bootstrapReadyCondition := conditions.Mirror(conditions.UnstructuredGetter(bootstrapConfig), clusterv1.BootstrapReadyCondition)
+	conditions.Set(m, bootstrapReadyCondition)
```

Comment:
Should this also fall back to using `external.IsReady(bootstrapConfig)` to support bootstrap providers that haven't adopted conditions yet?

Reply:
+1

Reply:
Implemented fall-back